### PR TITLE
chore: remove buffer-xor usage

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Remove dependency on unmaintained `buffer-xor`.

### CODE REVIEW

File: packages/state-transition/src/block/processRandao.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
-import xor from "buffer-xor";
 import {digest} from "@chainsafe/as-sha256";
 import {allForks} from "@lodestar/types";
 import {EPOCHS_PER_HISTORICAL_VECTOR} from "@lodestar/params";
 import {getRandaoMix} from "../util/index.js";
 import {verifyRandaoSignature} from "../signatureSets/index.js";
 import {CachedBeaconStateAllForks} from "../types.js";
 
+function xor(array1: Uint8Array, array2: Uint8Array): Uint8Array {
```
Comment (wemeetagain) [inline packages/state-transition/src/block/processRandao.ts:8]: can you copy this implementation? https://github.com/browserify/buffer-xor/blob/master/index.js

Comment (jeluard) [reply packages/state-transition/src/block/processRandao.ts:8]: Do you mean specifically the `Buffer.allocUnsafe` part? Rest is the same.
My understanding was that this call wasn't part of the critical path that required best performance, hence the move to a more generic `Uint8Array` usage.

Comment (wemeetagain) [reply packages/state-transition/src/block/processRandao.ts:8]: Yeah, that and the for loop condition.
This is part of the state transition, which is part of the critical path.

Comment (nflaig) [inline packages/state-transition/src/block/processRandao.ts:8]: calling those arrays is bit strange to me, maybe call them `b1` and `b2` (for bytes) or for simplicity just `a` and `b`

**Review by nflaig — COMMENTED** [review]
Are we comfortable to add this just like that?

I think we should at least have
- benchmarks that prove there is no performance regression, running those locally might be sufficient here
- some unit tests to ensure the function works as expected, hard to review otherwise

**Comment by jeluard** [comment]
After discussing with @wemeetagain we came to the conclusion that removing this dependency doesn't provide that much value.

**Pull request closed without merging**